Upstream-Status: Backport

* Backported from the ToT: https://crrev.com/c/1350983

Signed-off-by: Maksim Sisov <msisov@igalia.com>
---
From 795bc58118abea5fe4e974ec6ac27a39a6e5b99c Mon Sep 17 00:00:00 2001
From: Maksim Sisov <msisov@igalia.com>
Date: Thu, 29 Nov 2018 08:06:45 +0000
Subject: [PATCH 10/17] [ozone] move gbm wrapper into a separate
 source set.

This patch makes it possible to compile ozone/wayland
on systems without gbm available.

on a system without gbm available.

TEST: compile ozone/wayland with use_wayland_gbm = false
Change-Id: Ibda6632dcafe5ba0a46a819045b7f366d5c76a46
Reviewed-on: https://chromium-review.googlesource.com/c/1350983
Commit-Queue: Maksim Sisov <msisov@igalia.com>
Reviewed-by: Michael Spang <spang@chromium.org>
Cr-Commit-Position: refs/heads/master@{#612093}
---
 ui/ozone/common/linux/BUILD.gn                    | 15 +++++++++++++--
 ui/ozone/platform/drm/BUILD.gn                    |  8 +++++---
 ui/ozone/platform/wayland/BUILD.gn                |  5 ++---
 .../wayland/gpu/wayland_connection_proxy.h        |  2 +-
 4 files changed, 21 insertions(+), 9 deletions(-)

diff --git a/ui/ozone/common/linux/BUILD.gn b/ui/ozone/common/linux/BUILD.gn
index 3e9592901274..2bf3481ffe87 100644
--- a/ui/ozone/common/linux/BUILD.gn
+++ b/ui/ozone/common/linux/BUILD.gn
@@ -7,18 +7,29 @@ import("//ui/ozone/ozone.gni")
 
 assert(ozone_platform_gbm || ozone_platform_wayland)
 
-source_set("linux") {
+source_set("drm") {
   sources = [
     "drm_util_linux.cc",
     "drm_util_linux.h",
+  ]
+
+  deps = [
+    "//base:base",
+    "//build/config/linux/libdrm",
+    "//ui/gfx:buffer_types",
+  ]
+}
+
+source_set("gbm") {
+  sources = [
     "gbm_buffer.h",
     "gbm_device.h",
     "gbm_wrapper.cc",
   ]
 
   deps = [
+    ":drm",
     "//base:base",
-    "//build/config/linux/libdrm",
     "//third_party/minigbm",
     "//ui/gfx:buffer_types",
     "//ui/gfx:memory_buffer",
diff --git a/ui/ozone/platform/drm/BUILD.gn b/ui/ozone/platform/drm/BUILD.gn
index 490835ced411..5af8744a033b 100644
--- a/ui/ozone/platform/drm/BUILD.gn
+++ b/ui/ozone/platform/drm/BUILD.gn
@@ -3,8 +3,8 @@
 # found in the LICENSE file.
 
 import("//build/config/linux/pkg_config.gni")
-import("//ui/ozone/ozone.gni")
 import("//gpu/vulkan/features.gni")
+import("//ui/ozone/ozone.gni")
 
 visibility = [ "//ui/ozone/*" ]
 
@@ -144,7 +144,8 @@ source_set("gbm") {
     "//ui/gl",
     "//ui/ozone:ozone_base",
     "//ui/ozone/common",
-    "//ui/ozone/common/linux",
+    "//ui/ozone/common/linux:drm",
+    "//ui/ozone/common/linux:gbm",
     "//ui/ozone/public/interfaces",
     "//ui/platform_window",
   ]
@@ -191,7 +192,8 @@ source_set("gbm_unittests") {
     "//ui/gfx",
     "//ui/ozone:platform",
     "//ui/ozone/common",
-    "//ui/ozone/common/linux",
+    "//ui/ozone/common/linux:drm",
+    "//ui/ozone/common/linux:gbm",
   ]
 
   if (drm_commit_properties_on_page_flip) {
diff --git a/ui/ozone/platform/wayland/BUILD.gn b/ui/ozone/platform/wayland/BUILD.gn
index 4578b8005db7..d3d30fc87877 100644
--- a/ui/ozone/platform/wayland/BUILD.gn
+++ b/ui/ozone/platform/wayland/BUILD.gn
@@ -113,7 +113,7 @@ source_set("wayland") {
     "//ui/gfx/geometry",
     "//ui/ozone:ozone_base",
     "//ui/ozone/common",
-    "//ui/ozone/common/linux",
+    "//ui/ozone/common/linux:drm",
     "//ui/ozone/public/interfaces/wayland:wayland_interfaces",
     "//ui/platform_window",
     "//ui/platform_window/platform_window_handler",
@@ -136,9 +136,8 @@ source_set("wayland") {
 
     deps += [
       "//build/config/linux/libdrm",
-      "//third_party/minigbm",
       "//ui/gfx:memory_buffer",
-      "//ui/ozone/common/linux",
+      "//ui/ozone/common/linux:gbm",
     ]
   }
 
diff --git a/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h b/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h
index f06f9b9aadad..7c3d0725e5f4 100644
--- a/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h
+++ b/ui/ozone/platform/wayland/gpu/wayland_connection_proxy.h
@@ -15,7 +15,7 @@
 #include "ui/ozone/public/interfaces/wayland/wayland_connection.mojom.h"
 
 #if defined(WAYLAND_GBM)
-#include "ui/ozone/common/linux/gbm_device.h"
+#include "ui/ozone/common/linux/gbm_device.h"  // nogncheck
 #endif
 
 struct wl_shm;
-- 
2.17.1

